Feature/68/ci tests workflow

gluhovSemen opened the pull request:
Test running action added

gluhovSemen committed:
feat: dummy test and workflow for githab test action added

gluhovSemen committed:
fix: poetry debuging

gluhovSemen committed:
fix: poetry fix

gluhovSemen committed:
fix: hopfuly now poetry works

gluhovSemen committed:
fix

gluhovSemen committed:
fix 3

gluhovSemen committed:
fix 4

gluhovSemen committed:
feat: added test for django import

gluhovSemen committed:
fix 5

gluhovSemen committed:
fix 6

gluhovSemen committed:
fix 7

gluhovSemen committed:
fix 8

gluhovSemen committed:
feat: report added

gluhovSemen committed:
fix 8

gluhovSemen committed:
fix: finilized test action

egorvavilov commented on tests/dummy_test.py:
```diff
@@ -0,0 +1,11 @@
+# TODO this file will be deleted when actual tests will be ready.
+
+from manage import main
```
Lets create dummy test from django
https://docs.djangoproject.com/en/4.2/topics/testing/overview/

gluhovSemen replied:
done

JedrzejMaluszczak commented on .github/workflows/main.yaml:
```diff
@@ -0,0 +1,40 @@
+    steps:
+
+      - name: Check out repository
+        uses: actions/checkout@v3
```
why not v4?

gluhovSemen replied:
you are right, fixed

JedrzejMaluszczak commented on .github/workflows/main.yaml:
```diff
@@ -0,0 +1,40 @@
+
+      - name: Install Poetry
+        uses: snok/install-poetry@v1
+        with:
```
I thnik we should pass a poetry version

gluhovSemen replied:
done

gluhovSemen committed:
fix: poetry version and actios vertion added

gluhovSemen committed:
fix: poetry version fix

gluhovSemen committed:
fix: django tests are running now

gluhovSemen committed:
feat django test added

gluhovSemen committed:
fix: pytest deleted from poetry

gluhovSemen committed:
feat: secret key added

gluhovSemen committed:
feat: other deps added

gluhovSemen committed:
Merge remote-tracking branch 'origin/main' into feature/68/CI_tests_workflow

gluhovSemen committed:
fix test action after postgres added

feat: postgres image added

fix 9

fix 11

fix 12

fix 13

fix 14

fix 15

fix 16

fix 17

fix 18

fix 19

fix 20

fix 21

fix postgress connection

fix settings default

fix update the postgres container version

fix default values for DB deleted

fix var namings

fix more namming fixes

egorvavilov commented on .github/workflows/main.yaml:
```diff
@@ -5,6 +5,17 @@ on: [pull_request, push]
+
+    services:
+      postgres_main:
+        image: postgres:12
```
maybe let's use 14?
12 was released in 2019;)

gluhovSemen replied:
changed
